### PR TITLE
Use namespaced math functions

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
Use the C++ style math functions in the std namespace.

#### Testing
It should be fine as long as it builds and lgtm says it closes a bunch of warnings.

### CODE REVIEW

File: src/line.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ void bresenham( const point &p1, const point &p2, int t,
     const int sx = ( d.x == 0 ) ? 0 : sgn( d.x );
     const int sy = ( d.y == 0 ) ? 0 : sgn( d.y );
     // Absolute values of slopes x2 to avoid rounding errors.
-    const point a = abs( d ) * 2;
+    const point a = std::abs( d ) * 2;
```
Comment (kevingranade) [inline src/line.cpp:26]:
```suggestion
    const point a = abs( d ) * 2;
```

File: src/line.cpp [hunk 2]
```diff
@@ -264,7 +264,7 @@ float rl_dist_exact( const tripoint &loc1, const tripoint &loc2 )
 
 int manhattan_dist( const point &loc1, const point &loc2 )
 {
-    const point d = abs( loc1 - loc2 );
+    const point d = std::abs( loc1 - loc2 );
```
Comment (kevingranade) [inline src/line.cpp:267]:
```suggestion
    const point d = abs( loc1 - loc2 );
```

File: src/magic_spell_effect.cpp [hunk 3]
```diff
@@ -233,7 +233,7 @@ std::set<tripoint> spell_effect::spell_effect_line( const spell &, const tripoin
     // Clockwise Perpendicular of Delta vector
     const point delta_perp( -delta.y, delta.x );
 
-    const point abs_delta = abs( delta );
+    const point abs_delta = std::abs( delta );
```
Comment (ZhilkinSerg) [inline src/magic_spell_effect.cpp:236]: That is not in `std`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/e397b562bf40ee83a4fffdd6dde5d884f4f3e438/src/point.h#L314-L317

Comment (kevingranade) [inline src/magic_spell_effect.cpp:236]:
```suggestion
    const point abs_delta = abs( delta );
```

File: src/tileray.cpp [hunk 4]
```diff
@@ -26,7 +26,7 @@ tileray::tileray( int adir ): direction( adir )
 void tileray::init( const point &ad )
 {
     delta = ad;
-    abs_d = abs( delta );
+    abs_d = std::abs( delta );
```
Comment (kevingranade) [inline src/tileray.cpp:29]:
```suggestion
    abs_d = abs( delta );
```

File: src/tileray.cpp [hunk 5]
```diff
@@ -49,7 +49,7 @@ void tileray::init( int adir )
     float direction_radians = static_cast<float>( direction ) * M_PI / 180.0;
     rl_vec2d delta_f( cos( direction_radians ), sin( direction_radians ) );
     delta = ( delta_f * 100 ).as_point();
-    abs_d = abs( delta );
+    abs_d = std::abs( delta );
```
Comment (kevingranade) [inline src/tileray.cpp:52]:
```suggestion
    abs_d = abs( delta );
```

File: src/sdl_utils.h [hunk 6]
```diff
@@ -139,7 +139,7 @@ inline SDL_Color color_pixel_sepia( const SDL_Color &color )
     const Uint8 av = average_pixel_color( color );
     const float gammav = 1.6;
     const float pv = av / 255.0;
-    const Uint8 finalv = std::min( int( round( pow( pv, gammav ) * 150 ) ), 100 );
+    const Uint8 finalv = std::min( std::round( std::pow( pv, gammav ) * 150 ), 100.0 );
```
Comment (kevingranade) [inline src/sdl_utils.h:142]:
```suggestion
    const Uint8 finalv =
        std::min( static_cast<int>( std::round( std::pow( pv, gammav ) * 150 ) ), 100 );
```

**Comment by kevingranade** [comment]
Only 91? Well crap there's definitely something wrong.  It looks like lgtm isn't counting abs() or sqrt(), and maybe some others.

**Comment by kevingranade** [comment]
My guess is that these might actually be macros, which is annoying because AFAICT that means lgtm can't do checks based on their presence.

**Pull request merged**